.dotnet/src/Custom/VectorStores/VectorStoreFileBatchOperation.cs

```diff
@@ -0,0 +1,381 @@
+﻿using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Threading;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace OpenAI.VectorStores;
+
+/// <summary>
+/// Long-running operation for creating a vector store file batch.
+/// </summary>
+public partial class VectorStoreFileBatchOperation : OperationResult
+{
+    internal VectorStoreFileBatchOperation(
+        ClientPipeline pipeline,
+        Uri endpoint,
+        ClientResult<VectorStoreBatchFileJob> result)
+        : base(result.GetRawResponse())
+    {
+        _pipeline = pipeline;
+        _endpoint = endpoint;
+
+        Value = result;
+        Status = Value.Status;
+
+        _vectorStoreId = Value.VectorStoreId;
+        _batchId = Value.BatchId;
+
+        IsCompleted = GetIsCompleted(Value.Status);
+        RehydrationToken = new VectorStoreFileBatchOperationToken(VectorStoreId, BatchId);
+    }
+
+    /// <summary>
+    /// The current value of the <see cref="VectorStoreBatchFileJob"/> in progress.
+    /// </summary>
+    public VectorStoreBatchFileJob? Value { get; private set; }
+
+    /// <summary>
+    /// The current status of the <see cref="VectorStoreBatchFileJob"/> in progress.
+    /// </summary>
+    public VectorStoreBatchFileJobStatus? Status { get; private set; }
+
+    /// <summary>
+    /// The ID of the vector store corresponding to this batch file operation.
+    /// </summary>
+    public string VectorStoreId { get => _vectorStoreId; }
+
+    /// <summary>
+    /// The ID of the batch file job represented by this operation.
+    /// </summary>
+    public string BatchId { get => _batchId; }
+
+    /// <summary>
+    /// Recreates a <see cref="VectorStoreFileBatchOperation"/> from a rehydration token.
+    /// </summary>
+    /// <param name="client"> The <see cref="VectorStoreClient"/> used to obtain the 
+    /// operation status from the service. </param>
+    /// <param name="rehydrationToken"> The rehydration token corresponding to 
+    /// the operation to rehydrate. </param>
+    /// <param name="cancellationToken"> A token that can be used to cancel the 
+    /// request. </param>
+    /// <returns> The rehydrated operation. </returns>
+#pragma warning disable OPENAI001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
```

Comment:
This class should be part of the OPENAI001 warning.  If you add the `[Experimental("OPENAI001")]` attribute to it, does it let you remove these `#pragma` directives?